src/reconnect-strategy/mqtts.reconnect-strategy.ts

```diff
@@ -0,0 +1,15 @@
+export interface MqttsReconnectStrategy {
+    /**
+     * Whether reconnect should happen or not
+     */
+    should(reason?: any): boolean;
```

Comment:
I forgot, this should be more descriptive.

```suggestion
    shouldReconnect(reason?: any): boolean;
```

Reply:
I don't like methods containing more than 2 words. Especially when these words repeated like `reconnectStrategy.shouldReconnect`.
I agree `should` is not descriptive enough. I renamed it into a more descriptive `check(reason)`, sounds fine to me.

Reply:
Check is also fine